src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -83,18 +106,35 @@
 
 	private KafkaHeaderMapper headerMapper;
 
+	private RecordMessageConverter replyMessageConverter = new MessagingMessageConverter();
+
 	private MessageChannel sendFailureChannel;
 
 	private String sendFailureChannelName;
 
+	private MessageChannel sendSuccessChannel;
+
+	private String sendSuccessChannelName;
+
 	private ErrorMessageStrategy errorMessageStrategy = new DefaultErrorMessageStrategy();
 
+	private volatile boolean noOutputChannel;
+
 	public KafkaProducerMessageHandler(final KafkaTemplate<K, V> kafkaTemplate) {
 		Assert.notNull(kafkaTemplate, "kafkaTemplate cannot be null");
 		this.kafkaTemplate = kafkaTemplate;
+		this.isGateway = kafkaTemplate instanceof ReplyingKafkaTemplate;
```

Comment:
I think we need to mention this in the class JavaDocs.
Not so obvious how to create an Outbound Gateway.

Thanks